### PR TITLE
Tiny tables tests

### PR DESCRIPTION
**Pull request by GuutBoy**
Covers the TinyTables protocol suite with tests. 

As part of this work I threw in an optimization of how `TinyTablesSBool` and `TinyTablesElement` are constructed. Essentially, since these are immutable and both of these classes just wrap Boolean value we can use two static instances (for both classes) for all references, which we can get from a `getInstance` method rather than using the constructor.

Similarly for the immutable `TinyTablesTriple` class which is just a wrapper around three Boolean's. I.e., we can make do with just eight static instances.

### CODE REVIEW

**Comment by pffrandsen** [comment]
Sorry, I did the first part of the transition

**Comment by GuutBoy** [comment]
Can you say more specifically what needs to change to run on windows? I don’t have a windows machine handy at the moment.

**Comment by pffrandsen** [comment]
Hmm
It seems the rewrite I did yesterday fixed it. They run slow, but they do complete

**Comment by GuutBoy** [comment]
Great, so does that mean all your comments are addressed? If not what still needs to be done?

File: suite/tinytables/src/test/java/dk/alexandra/fresco/suite/tinytables/TestTinyTables.java [hunk 1]
```diff
@@ -144,114 +145,190 @@ public void removeFolder() throws IOException {
...
+    }
+  }
+
+  @Test(expected = UnsupportedOperationException.class)
+  public void testOpen() throws Throwable {
+    try {
+      runTest(new FieldBoolTests.TestOpen<>(), EvaluationStrategy.SEQUENTIAL_BATCHED, true,
+          "testOpen");
+      runTest(new FieldBoolTests.TestOpen<>(), EvaluationStrategy.SEQUENTIAL_BATCHED, false,
+          "tesOpen");
```
[the remark this answers is not in the document]
Comment (GuutBoy) [reply suite/tinytables/src/test/java/dk/alexandra/fresco/suite/tinytables/TestTinyTables.java]: Acutally, as we are just testing the offline phase, the second `runTest` should be redundant. Thanks for pointing it out.

File: suite/tinytables/src/main/java/dk/alexandra/fresco/suite/tinytables/ot/base/NetworkWrapper.java [hunk 2]
```diff
@@ -11,29 +13,40 @@
...
+  private boolean closed;
 
+  /**
+   * Wraps a FRESCO {@link CloseableNetwork} to implement a SCAPI  {@link Channel}.
+   * @param network a network, not this assumed to be not yet closed
+   * @param myId the id of this party
+   */
   public NetworkWrapper(Network network, int myId) {
-    this.network = network;
+    this.network =  network;
```
[the remark this answers is not in the document]
Comment (GuutBoy) [reply suite/tinytables/src/main/java/dk/alexandra/fresco/suite/tinytables/ot/base/NetworkWrapper.java]: Sure, good point.

File: suite/tinytables/src/main/java/dk/alexandra/fresco/suite/tinytables/datatypes/TinyTablesElement.java [hunk 3]
```diff
@@ -6,120 +6,125 @@
 
 /**
  * Instances of this class represents an additive share of a boolean value.
- *
- * @author Jonas Lindstrøm (jonas.lindstrom@alexandra.dk)
- *
  */
 public class TinyTablesElement implements Serializable {
```
[the remark this answers is not in the document]
Comment (GuutBoy) [reply suite/tinytables/src/main/java/dk/alexandra/fresco/suite/tinytables/datatypes/TinyTablesElement.java]: Yes, unfortunately it is needed for the OT library (SCAPI).

**Pull request merged**